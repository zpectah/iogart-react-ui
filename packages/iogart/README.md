# iogart-react-ui/iogart

Main components kit

## Scripts

## Contents
Name | Module | Status | Category | Documentation
--- | --- | --- | --- | ---
Action Bar | `ActionBar` | ❌ | Action | read more
Alert | `Alert` | ❌ | Feedback | read more
Autocomplete | `Autocomplete` | ❌ | Data | read more
Avatar | `Avatar` | ❌ | Content | read more
Badge | `Badge` | ❌ | Status | read more
Box | `Box` | ❌ | Layout | read more
Breadcrumbs | `Breadcrumbs` | ❌ | Navigation | read more
Button | `Button` | ❎ | Action | read more
ButtonGroup | `ButtonGroup` | ❌ | Action | read more
Card | `Card` | ❌ | Layout | read more
Checkbox | `Checkbox` | ❌ | Data | read more
CheckboxGroup | `CheckboxGroup` | ❌ | Data | read more
Code | `Code` | ❌ | Content | read more
Confirm Dialog | `ConfirmDialog` | ❌ | Feedback | read more
Container | `Container` | ❌ | Layout | read more
Data Table | `DataTable` | ❌ | Data | read more
Date Picker | `DatePicker` | ❌ | Data | read more
Date Time Picker | `DateTimePicker` | ❌ | Data | read more
Dialog | `Dialog` | ❌ | Feedback | read more
Divider | `Divider` | ❌ | Content | read more
Dropdown | `Dropdown` | ❌ | Data | read more
File | `File` | ❌ | Data | read more
Footer | `Footer` | ❌ | Layout | read more
Form | `Form` | ❌ | Data | read more
Grid | `Grid` | ❌ | Layout | read more
Header | `Header` | ❌ | Layout | read more
Heading | `Heading` | ❌ | Layout | read more
Help Text| `HelpText` | ❌ | Data | read more
Icon Button | `IconButton` | ❌ | Action | read more
Image | `Image` | ❌ | Content | read more
Label | `Label` | ❌ | Data | read more
Link | `Link` | ❌ | Action | read more
List | `List` | ❌ | Content | read more
Mark | `Mark` | ❌ | Status | read more
Menu | `Menu` | ❌ | Navigation | read more
Multiple Select | `MultipleSelect` | ❌ | Data | read more
Number | `Number` | ❌ | Data | read more
Password | `Password` | ❌ | Data | read more
Progress Bar | `ProgressBar` | ❌ | Status | read more
Progress Circle | `ProgressCircle` | ❌ | Status | read more
Radio | `Radio` | ❌ | Data | read more
Radio Group | `RadioGroup` | ❌ | Data | read more
Scrollable | `Scrollable` | ❌ | Layout | read more
Search | `Search` | ❌ | Data | read more
Select | `Select` | ❌ | Data | read more
Slider | `Slider` | ❌ | Data | read more
Stack | `Stack` | ❌ | Layout | read more
Switch | `Switch` | ❌ | Data | read more
Switch Group | `SwitchGroup` | ❌ | Data | read more
Table | `Table` | ❌ | Data | read more
Tabs | `Tabs` | ❌ | Navigation | read more
Tag | `Tag` | ❎ | Status | read more
Textarea | `Textarea` | ❌ | Data | read more
Text Field | `TextField` | ❌ | Data | read more
Time Picker | `TimePicker` | ❌ | Data | read more
Toast | `Toast` | ❌ | Feedback | read more
Toggle | `Toggle` | ❌ | Action | read more
Tooltip | `Tooltip` | ❌ | Feedback | read more
Typography | `Typography` | ❌ | Content | read more
Video | `Video` | ❌ | Content | read more




