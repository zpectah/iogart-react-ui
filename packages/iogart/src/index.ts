export * from './Button';
export * from './Tag';
export * from './Stack';
