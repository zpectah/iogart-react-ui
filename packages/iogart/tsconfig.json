{
  "extends": "../../shared/tsconfig.package.json",
  "compilerOptions": {
    "outDir": "../../dist/iogart"
  },
  "include": ["src"]
}
