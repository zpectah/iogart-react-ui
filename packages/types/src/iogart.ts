import { CSSProperties } from 'react';

import { SxProps } from './sx';

export interface IogartBaseUi {
  tabIndex?: number | undefined;
  className?: string | undefined;
  style?: CSSProperties | undefined;
}

export interface IogartCommonUi extends IogartBaseUi {
  sx?: SxProps | undefined;
}
