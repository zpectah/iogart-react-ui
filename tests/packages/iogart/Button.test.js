import renderer from 'react-test-renderer';

import { Button } from '@iogart-react-ui/iogart';

it('Button - onMouseEnter & onMouseLeave', () => {

    const component = renderer.create(
        <Button>Button</Button>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    renderer.act(() => {
        tree.props?.onMouseEnter();
    });

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    renderer.act(() => {
       tree.props?.onMouseLeave();
    });

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});
