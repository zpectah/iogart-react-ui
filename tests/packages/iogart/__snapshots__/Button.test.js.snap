// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button - onMouseEnter & onMouseLeave 1`] = `
<button
  className="iogart-base iogart-ButtonBase root-0-2-13 iogart-ui iogart-Button root-0-2-1"
  onBlur={[Function]}
  onClick={[Function]}
  onDoubleClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Button
</button>
`;

exports[`Button - onMouseEnter & onMouseLeave 2`] = `
<button
  className="iogart-base iogart-ButtonBase root-0-2-13 iogart--hover iogart-ui iogart-Button root-0-2-1"
  onBlur={[Function]}
  onClick={[Function]}
  onDoubleClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Button
</button>
`;

exports[`Button - onMouseEnter & onMouseLeave 3`] = `
<button
  className="iogart-base iogart-ButtonBase root-0-2-13 iogart-ui iogart-Button root-0-2-1"
  onBlur={[Function]}
  onClick={[Function]}
  onDoubleClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Button
</button>
`;
